jobs/JREGIONAL_RUN_PYTHON_ENSMEAN

```diff
@@ -0,0 +1,121 @@
+# PNG figures and zipping them for dissemination to the web.
+#
+#-----------------------------------------------------------------------
+#
+
+#
+#-----------------------------------------------------------------------
+#
+# Source the variable definitions file and the bash utility functions.
+#
+#-----------------------------------------------------------------------
+#
+. ${GLOBAL_VAR_DEFNS_FP}
+. $USHDIR/source_util_funcs.sh
+#
+#-----------------------------------------------------------------------
+#
+# Save current shell options (in a global array).  Then set new options
+# for this script/function.
+#
+#-----------------------------------------------------------------------
+#
+{ save_shell_opts; set -u -x; } > /dev/null 2>&1
+#
+#-----------------------------------------------------------------------
+#
+# Get the full path to the file in which this script/function is located 
+# (scrfunc_fp), the name of that file (scrfunc_fn), and the directory in
+# which the file is located (scrfunc_dir).
+#
+#-----------------------------------------------------------------------
+#
+scrfunc_fp=$( readlink -f "${BASH_SOURCE[0]}" )
+scrfunc_fn=$( basename "${scrfunc_fp}" )
+scrfunc_dir=$( dirname "${scrfunc_fp}" )
+#
+#-----------------------------------------------------------------------
+#
+# Print message indicating entry into script.
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "
+========================================================================
+Entering script:  \"${scrfunc_fn}\"
+In directory:     \"${scrfunc_dir}\"
+
+This is the J-job script for the task that runs the post-processor (UPP)
+on the output files corresponding to a specified forecast hour.
+========================================================================"
+#
+#-----------------------------------------------------------------------
+#
+# Create the working directory under the cycle directory.
+#
+#-----------------------------------------------------------------------
+#
+# CYCLE_TYPE is to indicate which type of cycle this job for:
+#   spinup: this job is for spin-up cycles
+#   others(prod): this job is for product cycle
+#
+CYCLE_TYPE=${CYCLE_TYPE:-prod}
+if [ ${CYCLE_TYPE} == "spinup" ]; then
+  modelinputdir=${CYCLE_DIR}/ensmean/fcst_fv3lam_spinup/INPUT
+else
+  modelinputdir=${CYCLE_DIR}/ensmean/fcst_fv3lam/INPUT
+fi
+if [ ! -d ${modelinputdir} ]; then
+  mkdir_vrfy -p ${modelinputdir}
+fi
+rm -f ${modelinputdir}/${FILETYPE}.nc
+#
+#-----------------------------------------------------------------------
+#
+# Call the driver script.
+#
+#-----------------------------------------------------------------------
+#
+cd_vrfy ${modelinputdir}
+
+mem0001_dir=${CYCLE_DIR}/mem0001/fcst_fv3lam/INPUT
+ln_vrfy  -snf  ${mem0001_dir}/coupler.res  coupler.res
+ln_vrfy  -snf  ${mem0001_dir}/fv_core.res.nc  fv_core.res.nc
+ln_vrfy  -snf  ${mem0001_dir}/gfs_ctrl.nc  gfs_ctrl.nc
```

Comment:
I'm a bit confused about what these links are doing. Maybe you can point directly to the files in the `-m` argument below?

Reply:
These files are not going through the ensemble mean calculation but are preparing for the next step - GSI analysis, to be consistent with the member analysis. They are needed for exregional_run_analysis.sh

Reply:
Those detailed actions should be in scripts level. J-Job only prepare environment and run directory.
You don't need to change it now if there are too much trouble to make those links inside the scripts. But we need to move those into scripts later.

Reply:
GSI script assumes these files are already in INPUT/ directory (bkpath) and decide whether it is cold start or warm start based on the availability of these files. Therefore I don't think it is a good idea to do the links in GSI analysis step.

Reply:
Ohhh. I am considering a few other issues with this now. 

What happens if you try, but fail at making an ensemble mean? Does the linking step overwrite data that used to be in those files? Are you left with Member 1 data in the `coupler.res`, etc files and can't easily go back to a different background in case you want to run without the ensemble mean (in case of failure). 

Perhaps it's safer to only link these files in the GSI step once you know for sure that the ensemble mean process gave you usable output?

Maybe the assumption in the GSI script should check on the existence of output from this step (ensemble mean creation) before it links these Member 1 files into the expected background location?